mono/mini/mini-generic-sharing.c

```diff
@@ -3856,6 +3856,8 @@ get_shared_type (MonoType *t, MonoType *type)
 	ttype = type->type;
 	if (type->type == MONO_TYPE_VALUETYPE) {
 		ttype = mono_class_enum_basetype_internal (type->data.klass)->type;
+	} else if (type->type == MONO_TYPE_GENERICINST && type->data.generic_class->container_class->enumtype) {
+		ttype = mono_class_enum_basetype_internal (mono_class_create_generic_inst (type->data.generic_class))->type;
 	} else if (MONO_TYPE_IS_REFERENCE (type)) {
```

Comment:
This should call mono_class_from_mono_type_internal () instead of mono_class_create_generic_inst ().